maven-core/src/main/java/org/apache/maven/lifecycle/providers/CleanLifecycleProvider.java

```diff
@@ -19,45 +19,33 @@
  * under the License.
  */
 
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-
 import javax.inject.Inject;
 import javax.inject.Named;
-import javax.inject.Provider;
 import javax.inject.Singleton;
 
-import org.apache.maven.lifecycle.Lifecycle;
-import org.apache.maven.lifecycle.mapping.LifecyclePhase;
-
+/**
+ * {@code clean} lifecycle provider.
+ */
 @Named( "clean" )
 @Singleton
 public final class CleanLifecycleProvider
-    implements Provider<Lifecycle>
+    extends AbstractLifecycleProvider
 {
-  private final Lifecycle lifecycle;
+    private static final String LIFECYCLE_ID = "clean";
 
-  @Inject
-  public CleanLifecycleProvider()
-  {
-    HashMap<String, LifecyclePhase> phases = new HashMap<>();
-    phases.put( "clean", new LifecyclePhase( "org.apache.maven.plugins:maven-clean-plugin:3.1.0:clean" ) );
-
-    this.lifecycle = new Lifecycle(
+    private static final String[] PHASES = {
+        "pre-clean",
         "clean",
-        Collections.unmodifiableList( Arrays.asList(
-                "pre-clean",
-                "clean",
-                "post-clean"
-        ) ),
-        Collections.unmodifiableMap( phases )
-    );
-  }
-
-  @Override
-  public Lifecycle get()
-  {
-    return lifecycle;
-  }
+        "post-clean"
+    };
+
+    private static final String[] BINDINGS = {
+        "clean", "org.apache.maven.plugins:maven-clean-plugin:3.1.0:clean"
+    };
+
+    @Inject
+    public CleanLifecycleProvider()
+    {
```

Comment:
Why `Inject` on constructor without arguments?
What should be injected ... ?

Reply:
Because it looks cool and professional

Reply:
Is explicit IMHO, also does not stirs anything (present or not present), so IMHO is okay.

Reply:
this part is not modified by the PR :)